frontend/src/components/design-system/containers/CardComponent.vue

```diff
@@ -0,0 +1,91 @@
+<script setup lang="ts">
+import { computed } from 'vue';
+
+export interface Props {
+  radius?: 'small' | 'large';
+  corner?: 'top-left' | 'top-right' | 'bottom-left' | 'bottom-right';
+  border?: boolean | 'muted';
```

Comment:
urcite prosim nekombinovat ruzny types v ramci jednoho enumu. neco jako  'default' | 'muted'; je podle me srozumitelnejsi. A zase ty optional props :)

Reply:
Hele ještě jsem nad tím přemýšlel a nechal bych to. Ta komponenta se pak bude hezky používat, napíšeš jenom <Card border>` a je to. Navíc nic jako border default nebo tak není definovaný, je prostě border a border strong.

Reply:
Za mě je lepší mít 'normal' a 'strong', než true/false/'strong'. S 'normal' a 'strong' se `classes` zjednodušší a budou lépe čítelné. Samotný border attribute bude také lépe pochopitelný. Teď to dává smysl, ale za nějaký čas na to jen budeš koukat a nebudeš vědět.

Reply:
Hele já fakt nevím. Podívej se na to CSS. Nebo pak teda dvě props, border boolean a borderKind normal/strong.

Reply:
Na co dva propsy? Stejným způsobem to děláš u ostatních propsů, tak proč tady to dělat jinak?

```
[`border--${props.border}`]: props.border,
```

```
.border {
  &--normal {
    border: var(--spacing-pico) solid rgb(var(--color-neutral-border));
  }

  &--strong {
    border-color: rgb(var(--color-neutral-border-strong));
  }

Reply:
No, tohle je sice řešení, ale není to DRY. Spíš teda takhle:

```js
'border': props.border,
[`border--${props.border}`]: props.border,
```

```css
.border {
  border-style: solid;
  border-width: var(--spacing-pico);

  &--normal {
    border-color: rgb(var(--color-neutral-border));
  }

  &--strong {
    border-color: rgb(var(--color-neutral-border-strong));
  }
}
```

Reply:
Sice jsi musel přidat jednu podmínku, takže jsi zase porušil jiný pravidlo. Ale nebudeme se tu hádat, který má větší prioritu (za mě jednoznačně WET :D). Takže to nebudeme řešit.

Každopádně v tomto řešení budeš muset overridnout `border-color`, protože default je na `currentColor`. V případě, že bys nechtěl border, tak bys jej přesto měl. A taky bys asi musel oddělat ty odečty 1px na `paddingu`, protože vždy bydeš mít okraj 1px širokej.

Reply:
Tak normal by byl default, takže k tomu, že by byla jenom .border by nikdy nedošlo. Jo, s tím paddingem to je good point, to mi nedošlo.

Reply:
Jak uz jsem rekl ja nemam rad kdyz se michaji ruzny typy u enumu takze ja jsem porad pro 'normal' | 'muted' nebo neco podobnyho, ale neprijde mi to jako nejakej showstopper.

Reply:
Tady se dohndnete vy dva @krystof-k a @Gebec a zbytek se podridi. Vy na to mate stronger opinion nez ja.

Reply:
Tak já to upravím podle toho posledního. Přijde mi to nejsystematičtější. Tam, kde je to není optional, tak bude vždycky jenom ta varianta.

Reply:
Hotovo, koukněte na to pls @Gebec @kyjanond.